test/services/courses/services/courseMembers.test.js

```diff
@@ -0,0 +1,167 @@
+const testObjects = require('../../helpers/testObjects')(app);
+const { generateRequestParamsFromUser } = require('../../helpers/services/login')(app);
+
+const courseMembersService = app.service('/courses/:scopeId/members');
+
+describe('course scope members service', () => {
+	let server;
+
+	before((done) => {
+		server = app.listen(0, done);
+	});
+
+	after((done) => {
+		server.close(done);
+	});
+
+	it('is properly registered', () => {
+		expect(courseMembersService).to.not.equal(undefined);
+	});
+
+	describe('in a small course', () => {
+		let teacher;
+		let student;
+		let course;
+
+		let teacherParams;
+		let studentParams;
+
+		before(async function before() {
+			this.timeout(5000);
+			teacher = await testObjects.createTestUser();
+			teacherParams = await generateRequestParamsFromUser(teacher);
+			student = await testObjects.createTestUser();
+			studentParams = await generateRequestParamsFromUser(student);
+			course = await testObjects.createTestCourse({
+				teacherIds: [teacher._id],
+				userIds: [student._id],
+			});
+		});
+
+		after(testObjects.cleanup);
+
+		it('returns the right members', async () => {
+			const response = await courseMembersService.find({ route: { scopeId: course._id } });
+			expect(Object.keys(response)).to.have.members([teacher._id.toString(), student._id.toString()]);
+		});
+
+		it('assigns the correct permissions to each userId', async () => {
+			const response = await courseMembersService.find({ route: { scopeId: course._id } });
+			for (const userId in response) {
+				if (userId === teacher._id.toString()) {
+					expect(response[userId]).to.include.members([
+						'COURSE_EDIT', 'COURSE_DELETE', 'SCOPE_PERMISSIONS_VIEW',
+					]);
+				} else if (userId === student._id.toString()) {
+					expect(response[userId]).to.include.members(['HOMEWORK_VIEW', 'COURSE_VIEW']);
+					expect(response[userId]).to.not.include.members([
+						'COURSE_EDIT', 'COURSE_DELETE', 'SCOPE_PERMISSIONS_VIEW',
+					]);
+				}
+			}
+		});
+
+		it('allows access to the teacher', async () => {
+			const params = {
+				...teacherParams,
+				route: { scopeId: course._id },
+			};
+			const response = await courseMembersService.find(params);
+			expect(response).to.not.equal(undefined);
+		});
+
+		it('does not allow access for students and other users', async () => {
+			const params = {
+				...studentParams,
+				route: { scopeId: course._id },
+			};
+			try {
+				await courseMembersService.find(params);
+				throw new Error('The previous call should have failed.');
+			} catch (err) {
+				expect(err).to.be.instanceOf(Forbidden);
+			}
+		});
+	});
+
+	describe('in a large course', () => {
+		let teachers;
+		let substitutionTeachers;
+		let students;
+		let course;
+
+		let teacherParams;
+		let substitutionTeacherParams;
+		let studentParams;
+
+		const toId = (item) => item._id;
+		const toIdString = (item) => item._id.toString();
+
+		before(async function before() {
+			this.timeout(8000);
```

Comment:
What is the different between small and large courses in test case? Should not pass every small course, if the large is success? I see only different between teachers and substitutions teachers. 
A real course can have 30 students if you want to test it. Maybe create students over loop?

Reply:
You're right. The course size is not important. I just wanted to have two scenarios to test if substitution teachers make a difference. `small` and `large` are maybe not the most acurate